internal/runtime/hcsv2/spec.go

```diff
@@ -61,7 +63,8 @@ func setUserStr(spec *oci.Spec, userstr string) error {
 			// evaluate username to uid/gid
 			return setUsername(spec, userstr)
 		}
-		return setUserID(spec, int(v))
+		spec.Process.User.UID, spec.Process.User.GID = uint32(v), 0
```

Comment:
Hmm. This means we won't get the gid even if the user is listed in the passwd file. Does that match what happens on Linux?

Reply:
It looks like containerd always tries to parse.

Reply:
Good catch. I will update this to first check if the user exists in /etc/passwd and use the gid in that case otherwise simply use 0 as the gid.

Reply:
Fixed.